src/daos-gen/daos-gen.c

```diff
@@ -0,0 +1,258 @@
+                usage = 1;
+                break;
+            case '?':
+                usage = 1;
+                break;
+            default:
+                if(rank == 0) {
+                    printf("?? getopt returned character code 0%o ??\n", c);
+                }
+        }
+    }
+
+    /* If we need to print the usage
+     * then do so before internal processing */
+    if (usage) {
+        if (rank == 0) {
+            print_usage();
+        }
+        mfu_finalize();
+        MPI_Finalize();
+        return 1;
+    }
+
+    int tmp_rc;
+
+    rc = daos_init();
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to initialize daos");
+        mfu_finalize();
+        MPI_Finalize();
+        return 1;
+    }
+
+    daos_obj_id_t oid;
+    daos_ofeat_t ofeats = DAOS_OF_DKEY_UINT64 | DAOS_OF_KV_FLAT | DAOS_OF_ARRAY;
+    /* connect to pool/cont then broadcast to rest of ranks */
+    if (rank == 0) {
+        /* generate container UUID */
+        uuid_generate(daos_args->src_cont_uuid);
+
+        daos_pool_info_t pool_info = {0};
+        daos_cont_info_t co_info = {0};
+#if DAOS_API_VERSION_MAJOR < 1
+        rc = daos_pool_connect(daos_args->src_pool_uuid, NULL, NULL, DAOS_PC_RW,
+                               &(daos_args->src_poh), &pool_info, NULL);
+#else
+        rc = daos_pool_connect(daos_args->src_pool_uuid, NULL, DAOS_PC_RW,
+                               &(daos_args->src_poh), &pool_info, NULL);
+#endif
+        if (rc != 0) {
+            MFU_LOG(MFU_LOG_ERR, "Failed to connect to pool: "DF_RC, DP_RC(rc));
+            rc = 1;
+        }
+
+        /* create cont and open */
+        rc = daos_cont_create(daos_args->src_poh, daos_args->src_cont_uuid, NULL, NULL);
+        if (rc != 0) {
+            MFU_LOG(MFU_LOG_ERR, "Failed to create cont: "DF_RC, DP_RC(rc));
+            rc = 1;
+        }
+        rc = daos_cont_open(daos_args->src_poh, daos_args->src_cont_uuid,
+                            DAOS_COO_RW, &daos_args->src_coh, &co_info, NULL);
+        if (rc != 0) {
+            MFU_LOG(MFU_LOG_ERR, "Failed to open container: "DF_RC, DP_RC(rc));
+            rc = 1;
+        }
+    }
+
+    /* broadcast rank 0's pool and cont handle to everyone else */
+    daos_bcast_handle(rank, &(daos_args->src_poh), &(daos_args->src_poh), POOL_HANDLE); 
+    daos_bcast_handle(rank, &(daos_args->src_coh), &(daos_args->src_poh), CONT_HANDLE); 
+
+	daos_handle_t	    oh;
+	daos_array_iod_t    iod;
+	d_sg_list_t         sgl;
+	daos_range_t	    rg;
+	d_iov_t		        iov;
+	char		        buf[BUFLEN], rbuf[BUFLEN];
+	daos_size_t	        array_size;
+
+    /* TODO: generate objects/data inside of container */
+    rc = daos_obj_generate_oid(daos_args->src_coh, &oid, ofeats, OC_SX, 0, 0);
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to generate oid: "DF_RC, DP_RC(rc));
+        rc = 1;
+    }
+
+	/** create the array */
+	rc = daos_array_create(daos_args->src_coh, oid, DAOS_TX_NONE, 1, 1048576, &oh,
```

Comment:
@dsikich We could probably allow configuring the chunk size and cell size through the command line